apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,134 @@
+	** SWR hook to get the expected out for a given in/out pair with a specific amount. This hook is
+	** called every 10s or when amount/in or out changes. Calls the allowanceFetcher callback.
+	**********************************************************************************************/
+	const	{data: vanillAllowanceFrom, isLoading: isValidatingAllowance, mutate: mutateAllowance} = useSWR(
+		canInteract && !isInputTokenEth && shouldUseVanillaAllowance ?
+			[selectedOptionFrom, spender] : null,
+		retrieveAllowance,
+		{revalidateOnFocus: false}
+	);
+
+	useEffect((): void => {
+		if (shouldUseVanillaAllowance && vanillAllowanceFrom) {
+			set_allowanceFrom(vanillAllowanceFrom);
+		}
+	}, [vanillAllowanceFrom, shouldUseVanillaAllowance]);
+
+	const onSuccess = useCallback(async (): Promise<void> => {
+		onChangeAmount(DefaultTNormalizedBN);
+		await refresh();
+	}, [onChangeAmount, refresh]);
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger an approve web3 action, simply trying to approve `amount` tokens
+	** to be used by the Partner contract or the final vault, in charge of
+	** depositing the tokens.
+	** This approve can not be triggered if the wallet is not active
+	** (not connected) or if the tx is still pending.
+	**************************************************************************/
+	async function	onApproveFrom(): Promise<void> {
+		onApprove(
+			set_txStatusApprove,
+			async (): Promise<void> => {
+				if (currentSolver === Solver.COWSWAP) {
+					set_allowanceFrom({
+						raw: ethers.constants.MaxUint256,
+						normalized: Infinity
+					});
+				} else {
+					await mutateAllowance();
+				}
+			}
+		);
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Wrapper to decide if we should use the partner contract or not
+	**************************************************************************/
+	if (
+		(currentSolver === Solver.VANILLA && (isDepositing &&( txStatusApprove.pending || amount.raw.gt(allowanceFrom?.raw || 0))))
+		|| (currentSolver === Solver.COWSWAP && (txStatusApprove.pending || amount.raw.gt(allowanceFrom?.raw || 0)))
+		|| (currentSolver === Solver.CHAIN_COIN && (!isDepositing && (txStatusApprove.pending || amount.raw.gt(allowanceFrom?.raw || 0))))
+		|| (currentSolver === Solver.PARTNER_CONTRACT && ((txStatusApprove.pending || amount.raw.gt(allowanceFrom?.raw || 0))))
+	) {
+		return (
+			<Button
+				className={'w-full'}
+				isBusy={txStatusApprove.pending || isValidatingAllowance}
+				isDisabled={!isActive || amount.raw.isZero() || (amount.raw).gt(maxDepositPossible.raw)}
+				onClick={onApproveFrom}>
+				{'Approve'}
+			</Button>
+		);
+	}
+
+	return (
+		<Button
+			onClick={async (): Promise<void> => (
+				isDepositing ?
+					onExecuteDeposit(set_txStatusDeposit, onSuccess) :
+					onExecuteWithdraw(set_txStatusDeposit, onSuccess)
```

Comment:
I didn't dig into this, but should this be something like `onExecuteWithdraw(set_txStatusWithdraw, onSuccess)`